### PR TITLE
Add git to the required tools

### PR DESCRIPTION
**Pull request by lf-**
### Description

Added git to the installed dependencies as it is an undocumented build dependency. I found this while building inside a Ubuntu docker container with the source bind-mounted into it ;-)

### Testing Done

It builds now.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Just wondering what the use case for this is. Most people's workflow is to check out the codebase using `git` and CI already has git

**Comment by lf-** [comment]
It's an undocumented dependency, and adding it is a no-op for most people who grabbed the source with git, but in some cases, such as the one I hit in a container with the source already bind mounted into it, where the build failed with an error during an actual build step rather than a precondition check, it is useful. Also if you download a tarball of the source and extract it, that doesn't require git (and is actually substantially faster, as I've learned working in the Nix ecosystem where this is commonplace).

**Comment by jonathanlew** [comment]
> Also if you download a tarball of the source and extract it, that doesn't require git (and is actually substantially faster, as I've learned working in the Nix ecosystem where this is commonplace).

That's really interesting. I'm wondering how that workflow works. Is this for development or just running the code in a docker? 

I have no problems with adding the dependency, just wanna know more about this

File: environment_setup/setup_software.sh [hunk 1]
```diff
@@ -45,18 +45,19 @@ host_software_packages=(
...
     python3       # Python 3
     python3-pip   # Required for bazel to install python dependencies for build targets
     python3-protobuf # This is required for the "NanoPb" library, which does not
-                    # properly manage this as a bazel dependency, so we have 
+                    # properly manage this as a bazel dependency, so we have
                     # to manually install it ourselves
     python3-yaml # yaml for cfg generation (Dynamic Parameters)
     qt5-default # The GUI library for our visualizer
     valgrind # Checks for memory leaks
+    git # required for build
```
Comment (jonathanlew) [inline environment_setup/setup_software.sh]: nit: this list _should_ be sorted alphabetically so please move it up a bit

**Comment by lf-** [comment]
Re: tarball, you'd possibly do this for CI but even moreso for dependencies that you don't need git metadata for as they're just used to build. It's just grabbing one of the archive download links with curl then extracting with tar. 

I mounted the code in from the host as a volume in a docker container so I can build it on Arch Linux, since our build system hardcodes so many paths it doesn't work at all on any non Ubuntu (due to where libc++.a is iirc) that's non gcc 7 (gcc 7 has its path hard coded). It really probably should use pkg-config to find its stuff. 

I was only building it to benchmark a build on some new hardware.

**Review by jonathanlew — APPROVED** [review]
weird CI timeout, restarted

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**